src/classes/UTIL_SimpleFieldMapper.cls

```diff
@@ -0,0 +1,282 @@
+    * @return Populated fields Map for Source Mapping Object
+    */
+    @testVisible private Map<String, Schema.SObjectField> getSourceFields() {
+        if (this.sourceFields == null) {
+            this.sourceFields = sourceSObjectType.getDescribe().fields.getMap();
+        }
+        return this.sourceFields;
+    }
+
+    /*********************************************************************************************************
+    * @description Returns a populated map of Target fields for the Target Mapping Object.
+    *              Note: Lazy, onetime initialization is used
+    * @return Populated fields Map for Target Mapping Object
+    */
+    @testVisible private Map<String, Schema.SObjectField> getTargetFields() {
+        if (this.targetFields == null) {
+            this.targetFields = targetSObjectType.getDescribe().fields.getMap();
+        }
+        return this.targetFields;
+    }
+
+    /*********************************************************************************************************
+    * @description Determines the Set of Mappable Fields for the two objects, filtering out
+    *              any that are incompatible, as well as ignoring ones that are present in the
+    *              fieldsToExclude set.  Note: Fields must be present with identical API names to
+    *              match, including the namespace prefix.
+    * @return Populated Set of Mappable fields for the Source and Target Object Types
+    */
+    public Set<String> determineMappableFields() {
+        Set<String> mappableFields = new Set<String>();
+        Set<String> candidateFields = new Set<String>();
+
+        // Will use whichever SObjectType has less fields to iterate over
+        Map<String, Schema.SObjectField> fieldsForReview = (getSourceFields().size() > getTargetFields().size()) ? getSourceFields() : getTargetFields();
+
+        for (String fieldForReview :fieldsForReview.keySet()) {
+            String candidateField = fieldForReview.toLowerCase();
+
+            if (this.fieldsToExclude.contains(candidateField)) { // Skip this field
+                continue;
+            }
+
+            Schema.SObjectField targetField = getTargetFields().get(candidateField);
+            Schema.SObjectField sourceField = getSourceFields().get(candidateField);
+
+            if (sourceField == null || targetField == null) { // Not a potential match
+                continue;
+            }
+
+            if (isFieldCandidateForMapping(candidateField, targetField, false)) {
+                candidateFields.add(candidateField);
+            }
+        }
+
+        // Check if candidate fields are compatible and then add to mappableFields Set
+        for (String candidateField : candidateFields) {
+            Schema.SObjectField sourceField = sourceFields.get(candidateField);
+            Schema.SObjectField targetField = targetFields.get(candidateField);
+
+            if (isCompatibleFieldMapping(sourceField, targetField)) {
```

Comment:
Can we avoid this loop and use `if (isCompatibleFieldMapping(sourceField, targetField))` in the above loop (line 183)?